### PR TITLE
Fix the arcchart rounded edges

### PR DESCRIPTION
**Pull request by Elv13**
Ref #2604

### CODE REVIEW

File: lib/gears/shape.lua [hunk 1]
```diff
@@ -472,8 +472,11 @@ function module.arc(cr, width, height, thickness, start_angle, end_angle, start_
     if end_rounded then
         arc_length = arc_length - thickness/2
 
-        -- And back to angles
-        end_angle = start_angle + (arc_length/(radius - thickness/2))
+        -- And back to angles. Also make sure to avoid underflowing when the
+        -- rounded edge radius is greater than the angle delta.
+        end_angle = math.max(
+            start_angle, start_angle + (arc_length/(radius - thickness/2))
```
Comment (psychon) [inline lib/gears/shape.lua]: Can this be rewritten as `end_angle = start_angle + math.max(0, ...)`? I was going to say "but angles describe a circle, after going around it once you are back to where you started" (and other complicated things). Before someone else is also confused, I'm suggesting this reformulation to make the intention here clearer. (Assuming I understood the intention correctly...)

Comment (psychon) [inline lib/gears/shape.lua]: Also, doesn't the subtraction around `start_angle` in line 469 need the same treatment?

Comment (psychon) [reply lib/gears/shape.lua]: I seem to have forgotten to press "comment" or something:

Doesn't the subtraction with `start_angle` that happened some lines above need the same treatment?

Comment (Elv13) [reply lib/gears/shape.lua]: > Doesn't the subtraction with start_angle that happened some lines above need the same treatment?

I tested with 0.001 and it rendered correctly. Small values may be a bit off where they "should be" but I think we can neglect this fact given it renders properly.

> 'm suggesting this reformulation to make the intention here clearer. (Assuming I understood the intention correctly...)

ok

**Pull request merged**